src/status_im/contexts/wallet/events/collectibles.cljs

```diff
@@ -0,0 +1,127 @@
+   :details          2
+   :community-header 3})
+
+(def fetch-type
+  {:never-fetch         0
+   :always-fetch        1
+   :fetch-if-not-cached 2
+   :fetch-if-cache-old  3})
+
+(def max-cache-age-seconds 3600)
+
+(rf/reg-event-fx
+ :wallet/request-collectibles
+ (fn [{:keys [db]} [{:keys [start-at-index new-request?]}]]
+   (let [request-id          0
+         collectibles-filter nil
+         data-type           (collectible-data-types :header)
+         fetch-criteria      {:fetch-type            (fetch-type :fetch-if-not-cached)
+                              :max-cache-age-seconds max-cache-age-seconds}
+         request-params      [request-id
+                              [(chain/chain-id db)]
+                              (keys (get-in db [:wallet :accounts]))
+                              collectibles-filter
+                              start-at-index
+                              collectibles-request-batch-size
+                              data-type
+                              fetch-criteria]]
+     {:fx [[:json-rpc/call
+            [{:method     "wallet_getOwnedCollectiblesAsync"
+              :params     request-params
+              :on-success (fn [])
+              :on-error   (fn [error]
+                            (log/error "failed to request collectibles"
+                                       {:event  :wallet/request-collectibles
+                                        :error  error
+                                        :params request-params}))}]]
+           (when new-request?
+             [:dispatch [:wallet/clear-stored-collectibles]])]})))
+
+(rf/reg-event-fx
+ :wallet/owned-collectibles-filtering-done
+ (fn [_ [{:keys [message]}]]
+   (let [{:keys [has-more offset
+                 collectibles]} (cske/transform-keys csk/->kebab-case-keyword (types/json->clj message))
+         start-at-index         (+ offset (count collectibles))]
+     {:fx [[:dispatch [:wallet/store-collectibles collectibles]]
+           (when has-more
+             [:dispatch [:wallet/request-collectibles {:start-at-index start-at-index}]])]})))
+
+(rf/reg-event-fx
+ :wallet/get-collectible-details
+ (fn [_ [collectible-id]]
+   (let [request-id               0
+         collectible-id-converted (cske/transform-keys csk/->PascalCaseKeyword collectible-id)
+         data-type                (collectible-data-types :details)
+         request-params           [request-id [collectible-id-converted] data-type]]
+     {:fx [[:json-rpc/call
+            [{:method   "wallet_getCollectiblesByUniqueIDAsync"
+              :params   request-params
+              :on-error (fn [error]
+                          (log/error "failed to request collectible"
+                                     {:event  :wallet/get-collectible-details
+                                      :error  error
+                                      :params request-params}))}]]]})))
+
+(rf/reg-event-fx
+ :wallet/get-collectible-details-done
```

Comment:
Not for this PR: Maybe we should use the suffix `-received` instead of `-done` as the collectibles and collectible details are received via signals.

Reply:
Yes, I think that's a better name. I think we should do it for new events :+1: